cmd/thv-operator/pkg/filtering/name_filter.go

```diff
@@ -0,0 +1,57 @@
+package filtering
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+// NameFilter handles name-based filtering using glob patterns
+type NameFilter interface {
+	// ShouldInclude determines if a server name should be included based on include/exclude patterns
+	// Returns (shouldInclude bool, reason string)
+	ShouldInclude(name string, include, exclude []string) (bool, string)
+}
+
+// DefaultNameFilter implements name filtering using Go's filepath.Match for glob patterns
+type DefaultNameFilter struct{}
+
+// NewDefaultNameFilter creates a new DefaultNameFilter
+func NewDefaultNameFilter() *DefaultNameFilter {
+	return &DefaultNameFilter{}
+}
+
+// ShouldInclude determines if a server name should be included based on include/exclude patterns
+//
+// Logic:
+// 1. If exclude patterns are specified and name matches any exclude pattern -> exclude (exclude takes precedence)
+// 2. If include patterns are specified and name matches any include pattern -> include
+// 3. If include patterns are specified and name doesn't match any -> exclude
+// 4. If only exclude patterns are specified (no include) and name doesn't match exclude -> include
+// 5. If no patterns are specified -> include (default behavior)
+func (*DefaultNameFilter) ShouldInclude(name string, include, exclude []string) (bool, string) {
+	// Check exclude patterns first (exclude takes precedence)
+	if len(exclude) > 0 {
+		for _, pattern := range exclude {
+			if matches, _ := filepath.Match(pattern, name); matches {
```

Comment:
I guess the reason filepath.Match error is ignored is that we'd just log it anyway?